raiden/utils/upgrades.py

```diff
@@ -115,18 +135,30 @@ def run(self):
         all data to the current version's database and execute the migration
         functions.
         """
-        old_db_filename = older_db_file(str(self._current_db_filename.parent))
+        paths = glob(f'{self._current_db_filename.parent}/v*_log.db')
+        old_db_filename = older_db_file(paths)
 
         if old_db_filename is None:
             return
 
         with get_file_lock(old_db_filename), get_file_lock(self._current_db_filename):
-            if get_db_version(self._current_db_filename) == RAIDEN_DB_VERSION:
-                # The current version has already been created / updraded.
-                return
-            else:
-                # The version inside the current database was not the expected one.
-                # Delete and re-run migration
+            if self._current_db_filename.exists():
+                db_version = get_db_version(self._current_db_filename)
+
+                # The current version matches our target version, nothing to do.
+                if db_version == RAIDEN_DB_VERSION:
+                    return
+
+                if db_version > RAIDEN_DB_VERSION:
```

Comment:
This is a different special case that was not handled before, this should never happen.